kgforge/specializations/models/rdf/service.py

```diff
@@ -128,16 +141,22 @@ def lookup_shape_from_node(self, node: URIRef) -> Shape:
 
 class RdfService:
 
-    def __init__(self, graph: Graph, context_iri: Optional[str] = None) -> None:
+    def __init__(self, graph: GraphLike, context_iri: Optional[str] = None) -> None:
 
         if context_iri is None:
             raise ConfigurationError("RdfModel requires a context")
         self._graph = graph
+        self.NXV = Namespace("https://bluebrain.github.io/nexus/vocabulary/")
```

Comment:
this is very Nexus specific, I am not convinced it should be part of the RdfService

Reply:
It is a namespace defined and used by Nexus Delta for its vocab (NXV.deprecated, NXV.createdBy, NXV.shapes, ...). This PR aligned the format for shacl expected by all rdf services. There is now a need for an id of type NXV.Schema which can point to one or many shacl shapes through NXV.shapes. So while NXV namespace was defined by Nexus Delta in the first place, it is here just a namespace needed to define schemas. It's usage does not implies using BlueBrainNexusStore.

Reply:
But any namespace used ... it should not be something hard-coded, it should be something configurable, no?

Reply:
Not always I might say. There is a balance to find. NXV.shapes or OWL.imports in [here](https://github.com/BlueBrain/nexus-forge/pull/393/files#diff-20fe56168eb41def80c78aa3d4c1f0dbeddb920f8859a795c86f3f148cc4c408R302) can't be changed (for the servie to work) removing the necessity to configure that for now. If in the future we want to support use defined schema format in addition to shacl, the configuration would definitely make sense

Reply:
Can't this be propagated from the Store/vocabulary/namespace config entry?

Reply:
I don't understand why the code you've pointed to is not compatible with a namespace provided on service initialisation

Reply:
Variables are added to config is there is a need to change them and use different values. As of now, the schema data model impose to use OWL.imports, NXV.shapes. So those are not subject to configuration and are fix values. The implementation can evolve though to enable use define schema format but not just yet: when this happen then they ca go to config file

Reply:
Yeah, seems like the current RDF model it's already an specialization, it would be nice to make a real `base` class

Reply:
Once we have another non demo model we can evaluate the need for a base non core model class